addons/web/models/models.py

```diff
@@ -371,6 +369,26 @@ def formatted_read_group(self, domain, groupby=(), aggregates=(), having=(), off
 
         return self._web_read_group_format(groupby, aggregates, groups)
 
+    def _get_currency_field_agg(self, aggregate):
+        fname = aggregate.split(':')[0].split('.')[0]
+        field = self._fields.get(fname)
+        if field and field.type == 'monetary':
+            return f'{field.get_currency_field(self)}:count_distinct'
+
+        return None
```

Comment:
I think it will be cleaner to have a method that returning a dict {monetary_aggregate: currency_aggregate_related}. Something like that:

```suggestion
    def _get_mapping_currency_aggregates(self, aggregates):
        return {
            aggregate: f'{field.get_currency_field(self)}:count_distinct'
            for aggregate in aggregates
            if (field := self._fields.get(aggregate.split(':')[0].split('.')[0]))
            if field.type == 'monetary'
        }
```